src/decoder/mod.rs

```diff
@@ -1089,11 +1098,17 @@ impl<R: Read + Seek> Decoder<R> {
         self.goto_offset_u64(offset)?;
 
         let line_samples = output_width * self.image().bits_per_sample.len();
-        let compression_method = self.image().compression_method;
         let photometric_interpretation = self.image().photometric_interpretation;
+        let compression_method = self.image().compression_method;
+        let jpeg_tables = self.image().jpeg_tables.clone();
```

Comment:
This could be a potentially large allocation, right? Could we use an Arc to avoid that?

Reply:
I have now changed it to pass an Arc<Vec<u8>> instead